### PR TITLE
Issues #705 and #706

### PR DESCRIPTION
**Pull request by ASL-rmarshall**
### Issue #705:
- Changed `record_count.py` to remerge results of a filtered grouped count onto the full list of (unfiltered) groups, filling any blanks with zero.
-   Updated documentation in `Operation.md` as requested:
    - Changed single quotes to backticks for parameter names
    - Corrected "values(s)" typos
    - Added examples

### Issue #706:
- Updated `Operations.json` schema to add `group_aliases` as an optional property (as a copy of `group`).
- Added `grouping_aliases` as a new operation parameter in `operation_params.py`.
- Mapped `group_aliases` to `grouping_aliases` in `rule_processor.py`.
- Added new methods to `base_operation.py`:
    - `_rename_grouping_columns` to rename each grouping column in `grouping` with the column name in the corresponding position in `grouping_aliases`.
    - `_get_grouping_columns` to create a list of grouping column names depending on whether `grouping_aliases` is specified or not.
- Updated documentation for both `distinct` and `record_count` in `Operation.md` to describe use of `group_aliases`.
- Created new `_replace_nans_in_specified_cols_with_none` static method in `base_data_service.py`.
- Updated `_handle_grouped_results` in `base_operation.py` to:
    - Include a call to `_rename_grouping_columns` if the `grouping_aliases` operation parameter is used (in addition to `grouping`).
    - Handle renamed grouping columns when merging the grouped results back onto the evaluation dataset.
    - Include a call to `_replace_nans_in_specified_cols_with_none` so that the `NaN` values created by the left join (when there is no grouped result for a particular group) do not cause an invalid JSON error when returned to the rules editor.
- Updated `test_grouped_distinct` in `test_distinct.py` to:
    - Add `grouping_aliases` parameter
    - Include an additional set of parameters that includes specification of `grouping_aliases`.
    - Update logic to test for correct application of `grouping_aliases`.
- Updated all tests in `test_record_count.py`:
    - `test_filtered_record_count`:
        - Moved filter specification to a parameter
        - Added a parameter set to confirm expected reporting of zero records matching filter.
    - `test_grouped_record_count`:
        - Added `grouping_aliases` parameter
        - Included a parameter set with a single-column `grouping_aliases`
        - Updated logic to test for correct application of `grouping_aliases`
    - `test_multi_group_record_count`:
        - Added `grouping_aliases` parameter
        - Included parameter sets with a `grouping_aliases` containing:
            - The same number of columns as `grouping`
            - Few columns than `grouping`
            - More columns than `grouping`
        - Updated logic to test for correct application of `grouping_aliases`
    - `test_filtered_grouped_record_count`:
        - Added `grouping_aliases` parameter
        - Included parameter set with:
            - A single-column `grouping_aliases`
            - Sets of values to test where:
                - There are no records matching the filter for an existing group (expecting zero to be reported as per #705 update)
                - The `grouping_aliases` column contains a grouping value not in `grouping` (expected `None` to be reported)
        - Updated logic to test for correct application of `grouping_aliases`

### Additional changes:
- Corrected more typos in `Operations.md`:
    - Un-indented `Operations` in the example for `distinct`
    - Removed trailing backticks from lines in yaml block code in examples for `label_referenced_variable_metadata` and `required_variables`

### CODE REVIEW

File: resources/schema/Operations.md [hunk 1]
```diff
@@ -515,7 +517,62 @@ Returns the permissible variables for a given domain and standard
...
+
+If `group` is provided, `group_aliases` may also be provided to assign new grouping variable names so that results grouped by the values in one set of grouping variables can be merged onto a dataset according to the same grouping value(s) stored in different set of grouping variables. When both `group` and `group_aliases` are provided, columns are renamed according to corresponding list position (i.e., the 1st column in `group` is renamed to the 1st columnin `group_aliases`, etc.). If there are more columns listed in `group` than in `group_aliases`, only the `group` columns with corresponding `group_aliases` columns will be renamed. If there are more columns listed in `group_aliases` than in `group`, the extra column names in `group_aliases` will be ignored.
```
Comment (SFJohnson24) [inline resources/schema/Operations.md:522]: I believe there is a space missing here at columnin

Comment (ASL-rmarshall) [reply resources/schema/Operations.md]: Typo corrected here and on [line 43](https://github.com/cdisc-org/cdisc-rules-engine/blob/706-allow-renaming-of-merge-key-for-operations-with-group/resources/schema/Operations.md?plain=1#L43)

File: resources/schema/Operations.md [hunk 2]
```diff
@@ -515,7 +517,62 @@ Returns the permissible variables for a given domain and standard
...
+```yaml
+- operation: record_count
+  id: $flagged_records_per_usubjid
+  group:
+    - USUBJID
+  filter:
+    FLAGVAR: "Y"
+```
+
+Example: return the number of records grouped by USUBJID and IDVARVAL where QNAM = "TEST1" and "IDVAR" = "GROUPID", renaming the IDVARVAL column to GROUPID for subsequent merging.
```
[the remark this answers is not in the document]
Comment (ASL-rmarshall) [reply resources/schema/Operations.md]: Quotes removed from IDVAR.

**Comment by ASL-rmarshall** [comment]
@SFJohnson24 I've corrected the "columnin" typos (the same text is used for both `distinct` and `record_count`) and the "IDVAR" quoting.

I'd also re-based the branch for some additional testing at my end - hence the additional commits. I hope that hasn't messed anything up.

**Review by SFJohnson24 — APPROVED** [review]
see first review comment, grammatical errors have been corrected.  Nice work

**Pull request merged**